client/src/Components/Location/Location.js

```diff
@@ -2,18 +2,45 @@ import React, { Component } from 'react';
 import './Location.css';
 
 class Location extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {wifi: false, privacy: false, quiet: false, foodAndDrink: false};
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    this.setState({
+      [event.target.id]: !this.state[event.target.id]
```

Comment:
Since the state is an object as well it references the value stored with the key given by the event target id. So if the checkbox clicked was wifi, event.target.id is wifi and it will set the state's wifi to be the opposite of whatever it was before